### ISSUE TITLE
Update the Contributor Discussions page

### ISSUE THREAD

**Issue by akatsoulas** [mozilla/sumo#1626]
See [figma design](https://www.figma.com/file/T7fMC8k01Lrbp7zHknBvx3/Information-Architecture?type=design&node-id=389-3194&mode=design&t=ed7FyS3IELLCGw1l-4)

**Comment by akatsoulas** [comment]
PR https://github.com/mozilla/kitsune/pull/5975

**Comment by emilghittasv** [comment]
This ticket is verified in `dev`.

* The "Contributor Forums" page has been renamed to "Contributor Discussions".
* The "For Contributors" side navbar has been renamed to "Contributor Discussions".
* The "Contributor Discussions" side navbar options have been updated to include only the discussion boards & are redirecting the user to the correct pages.
* The "Contributor Discussions" side navbar contains the correct elements in the following pages as well:
`/forums/contributors/`, `/forums/l10n-forum/`, `/forums/social-media/`,`/kb/all/discussions`,`/forums/mobile-support/`,`/forums/knowledge-base-articles/`,`/forums/forum-moderators/`,`/forums/off-topic/`,`/forums/lost-threads/` and `/forums/social-media/`


During the verification of this ticket I've noticed that the "Social Support Discussions" option from the "Contributor Discussions" side navbar is duplicated. Tracked under -> https://github.com/mozilla/sumo/issues/1751

I'm going to keep this ticket under the `QA` column until the IA work is pushed to `stage` and the verification is redone in that environment as well.
